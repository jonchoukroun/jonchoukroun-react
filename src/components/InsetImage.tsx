import React, { FC } from "react";

import styled, { css } from "styled-components";

interface IPictureProps {
    srcUrl?: string;
    width?: number | string;
    height?: number | string;
    textOverlay?: string;
}

const PictureContainer = styled.div`
    ${({ theme }) => css`
        box-shadow: -2px -2px 0px ${theme.colors.darkGray},
            2px 2px 0px ${theme.colors.white};

        /* Temp */
        background: linear-gradient(
            ${theme.colors.white},
            ${theme.colors.blue}
        );
        text-shadow: 1px 1px 0px rgba(255, 255, 255, 0.1),
            -1px -1px 0px rgba(0, 0, 0, 0.5);
        color: transparent;
        padding: 100px 25px;
        text-align: center;

        svg: {
            width: 100%;
        }
    `}
`;

const InsetImage: FC<IPictureProps> = ({
    srcUrl,
    width,
    height,
    textOverlay,
}) => {
    const pictureStyle = {
        width: width || "auto",
        height: height || "auto",
    };
    return (
        <PictureContainer style={pictureStyle}>
            <svg viewBox="0 0 300 24">
                <text textLength="290" lengthAdjust="spacing" x="5" y="15">
                    {textOverlay}
                </text>
            </svg>
        </PictureContainer>
    );
};

export default InsetImage;
