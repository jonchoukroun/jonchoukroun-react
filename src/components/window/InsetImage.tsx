import React, { FC } from "react";

import styled, { css } from "styled-components";

interface IPictureProps {
    srcUrl?: string;
    width?: number | string;
    height?: number | string;
}

const PictureContainer = styled.div`
    ${({ theme }) => css`
        min-height: 200px;
        box-shadow: -2px -2px 0px ${theme.colors.darkGray},
            2px 2px 0px ${theme.colors.white};

        /* Temp */
        background: linear-gradient(
            ${theme.colors.white},
            ${theme.colors.blue}
        );
        font-size: 85px;
        text-shadow: 1px 1px 0px rgba(255, 255, 255, 0.1),
            -1px -1px 0px rgba(0, 0, 0, 0.5);
        color: transparent;
        padding: 100px 25px;
        text-align: center;
    `}
`;

const InsetImage: FC<IPictureProps> = ({ srcUrl, width, height }) => {
    const pictureStyle = {
        width: width || "auto",
        height: height || "auto",
    };
    return (
        <PictureContainer style={pictureStyle}>Jon Choukroun</PictureContainer>
    );
};

export default InsetImage;
