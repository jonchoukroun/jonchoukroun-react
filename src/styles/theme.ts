export default {
    colors: {
        green: "#008080",
        blue: "#000080",
        black: "#141414",
        lightGray: "#f8f8f8",
        gray: "#c0c0c0",
        darkGray: "#878787",
        white: "#fbfbfb",
    },
    header: {
        fontWeight: "500",
        transform: "scaleX(1.1),",
    },
    body: {
        fontSize: "18px",
    },
    border: {
        line: "2px solid",
        color: "#f8f8f8 #7f787f #7f787f #f8f8f8",
        radius: "0",
    },
    defaultDropShadow: "1px 2px 0px 1px rgba(20, 20, 20, 0.7)",
    darkDropShadow: "1px 1px 0px 3px rgba(0, 0, 0, 0.7)",
};
