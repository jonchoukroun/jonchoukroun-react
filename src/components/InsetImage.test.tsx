import React from "react";
import InsetImage from "./InsetImage";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import theme from "../styles/theme";
import { ThemeProvider } from "styled-components";

it("renders successfully", () => {
    shallow(
        <ThemeProvider theme={theme}>
            <InsetImage />
        </ThemeProvider>
    );
});

it("renders an image", () => {
    const component = renderer.create(
        <ThemeProvider theme={theme}>
            <InsetImage />
        </ThemeProvider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
