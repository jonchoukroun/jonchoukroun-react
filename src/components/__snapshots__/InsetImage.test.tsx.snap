// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an image 1`] = `
<div
  className="sc-AxjAm iELqXV"
  style={
    Object {
      "height": "auto",
      "width": "auto",
    }
  }
>
  <svg
    viewBox="0 0 300 24"
  >
    <text
      lengthAdjust="spacing"
      textLength="290"
      x="5"
      y="15"
    />
  </svg>
</div>
`;
